diff --git a/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotification.java b/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotification.java
index 4ceb65d..567f0ea 100644
--- a/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotification.java
+++ b/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotification.java
@@ -34,8 +34,6 @@ import android.util.Log;

 import com.google.android.gms.tasks.OnCompleteListener;
 import com.google.android.gms.tasks.Task;
-import com.google.firebase.iid.FirebaseInstanceId;
-import com.google.firebase.iid.InstanceIdResult;
 import com.google.firebase.messaging.FirebaseMessaging;

 public class RNPushNotification extends ReactContextBaseJavaModule implements ActivityEventListener {
@@ -127,17 +125,17 @@ public class RNPushNotification extends ReactContextBaseJavaModule implements Ac
     public void requestPermissions() {
       final RNPushNotificationJsDelivery fMjsDelivery = mJsDelivery;

-      FirebaseInstanceId.getInstance().getInstanceId()
-              .addOnCompleteListener(new OnCompleteListener<InstanceIdResult>() {
+      FirebaseMessaging.getInstance().getToken()
+              .addOnCompleteListener(new OnCompleteListener<String>() {
                   @Override
-                  public void onComplete(@NonNull Task<InstanceIdResult> task) {
+                  public void onComplete(@NonNull Task<String> task) {
                       if (!task.isSuccessful()) {
                           Log.e(LOG_TAG, "exception", task.getException());
                           return;
                       }

                       WritableMap params = Arguments.createMap();
-                      params.putString("deviceToken", task.getResult().getToken());
+                      params.putString("deviceToken", task.getResult());
                       fMjsDelivery.sendEvent("remoteNotificationsRegistered", params);
                   }
               });
@@ -271,9 +269,9 @@ public class RNPushNotification extends ReactContextBaseJavaModule implements Ac
           @Override
           public void run() {
               try {
-                  FirebaseInstanceId.getInstance().deleteInstanceId();
+                  FirebaseMessaging.getInstance().deleteToken();
                   Log.i(LOG_TAG, "InstanceID deleted");
-              } catch (IOException e) {
+              } catch (Exception e) {
                   Log.e(LOG_TAG, "exception", e);
               }
           }
